src/egui/mod.rs

```diff
@@ -308,32 +309,33 @@ impl UpdatesApp {
                 merge.last_received_status = status;
             }
 
-            let result = match merge.promise.ready() {
-                Some(res) => res,
-                None => continue
-            };
-
-            match result {
-                Ok(_) => {
-                    info!("Merge completed for {}", &merge.title_id);
+            if let Some(result) = merge.promise.ready() {
+                match result {
+                    Ok(_) => {
+                        info!("Merge completed for {}", &merge.title_id);
 
-                    toasts.push((format!("{} merged successfully!", &merge.title_id), ToastLevel::Success));
-                    self.v.completed_merges.push(merge.title_id.clone());
-                }
-                Err(e) => {
-                    self.v.failed_merges.push(merge.title_id.clone());
+                        toasts.push((format!("{} merged successfully!", &merge.title_id), ToastLevel::Success));
+                        self.v.completed_merges.push(merge.title_id.clone());
+                    }
+                    Err(e) => {
+                        self.v.failed_merges.push(merge.title_id.clone());
 
-                    match e {
-                        MergeError::FilepathMismatch(_) | MergeError::PackagesUnmergable(_) | MergeError::FileMergeFailure => {
-                            toasts.push((format!("Failed to merge {}. Check the log for details.", merge.title_id), ToastLevel::Error));
+                        match e {
+                            MergeError::FilepathMismatch(_) | MergeError::PackagesUnmergable(_) | MergeError::FileMergeFailure => {
+                                toasts.push((format!("Failed to merge {}. Check the log for details.", merge.title_id), ToastLevel::Error));
+                            }
                         }
-                    }
 
-                    error!("Could not merge files for {}, reason: {:?}", merge.title_id, e);
+                        error!("Could not merge files for {}, reason: {:?}", merge.title_id, e);
+                    }
                 }
+
+                finished_merge_indexes.push(i);
             }
+        }
 
-            self.v.merge_queue.remove(i);
+        for (_,idx) in finished_merge_indexes.iter().enumerate() {
```

Comment:
You should iterate through this from the last element to the first, otherwise the indexes will get shifted as you remove items. Also, unless I'm missing something, I don't think you are making use of the enumerate call here right?
```suggestion
        for (_,idx) in finished_merge_indexes.iter().rev() {
```

Reply:
Yeah enumerate is a mistake here. I had noticed I used the wrong method while fixing but because in a hurry I forgot first to commit changes to one of the comments (changing match to result) but realized that mistake and pushed it after making comments, but now after waking up I see that I also pushed the version with wrong method. Sorry